src/rgw/rgw_redis_common.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <boost/asio/detached.hpp>
+#include <iostream>
+#include <string>
+
+#include "boost/redis/connection.hpp"
+#include "common/async/blocked_completion.h"
+#include "common/async/yield_context.h"
+#include "common/dout.h"
+
+namespace rgw {
+namespace redis {
+
+using boost::redis::config;
+using boost::redis::connection;
+using RedisResponseMap =
+    boost::redis::response<std::map<std::string, std::string>>;
+
+// struct ReadResponse {
+//   int errorCode;
+//   std::string errorMessage;
+
+//   ReadResponse(RedisResponseMap resp) {
+//     errorCode = std::stoi(std::get<0>(resp).value()["errorCode"]);
+//     errorMessage = std::get<0>(resp).value()["errorMessage"];
+//   }
+// };
+
+struct RedisResponse {
+  int errorCode;
+  std::string errorMessage;
+  std::string data;
+
+  RedisResponse(int ec, std::string msg)
+      : errorCode(ec), errorMessage(msg), data("") {}
+
+  RedisResponse(RedisResponseMap resp) {
+    errorCode = std::stoi(std::get<0>(resp).value()["errorCode"]);
```

Comment:
can you move these calls to the init line (similar to the other ctor)?
if so, you can make all data members `const`